src/implements.jl

```diff
@@ -40,14 +40,14 @@ using BaseInterfaces
 @implements BaseInterfaces.IterationInterface{(:indexing,:reverse)} MyObject
 ```
 """
-macro implements(interface, objtype)
-    _implements_inner(interface, objtype)
+macro implements(interface, objtype, test_objects)
+    _implements_inner(interface, objtype, test_objects)
 end
-macro implements(dev::Symbol, interface, objtype)
+macro implements(dev::Symbol, interface, objtype, test_objects)
     dev == :dev || error("3 arg version of `@implements must start with `dev`, and should only be used in testing")
-    _implements_inner(interface, objtype; show=true)
+    _implements_inner(interface, objtype, test_objects; show=true)
 end
-function _implements_inner(interface, objtype; show=false)
+function _implements_inner(interface, objtype, test_objects; show=false)
```

Comment:
So the test does _not_ happen when we run `@implements`?

Reply:
No, it just puts the expression to generate the objects into `test_objects(T, O) = $test_objects` for us to get later